client/react/component/ClientReferrals.js

```diff
@@ -0,0 +1,282 @@
+import { createContainer } from 'meteor/react-meteor-data';
+import { Clients } from '../../../imports/api/clients/clients';
+import { logger } from '/imports/utils/logger';
+
+// console.log(ReferralStatusList);
+
+class ClientReferrals extends React.Component {
+  constructor(props) {
+    super(props);
+
+    // This binding is necessary to make `this` work in the callback
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  getLastStatus() {
+    if (this.props.loading) {
+      return '';
+    }
+    const client = this.props.client;
+    // console.log(client);
+    // console.log("client information");
+    if (client.referralStatusHistory) {
+      return client.referralStatusHistory[client.referralStatusHistory.length - 1];
+    }
+    else {
+      return 0;
+    }
+  }
+
+  getStatusTooltip(step) {
+    if (this.props.loading) {
+      return '';
+    }
+    const client = this.props.client;
+    let history = ReferralStatusList[step].desc;
+    (client.referralStatusHistory || []).forEach((item) => {
+      if (item.status === step) {
+        const txt = item.statusDescription || item.comments;
+        history = `${history}<br />${item.dateUpdated} - ${txt}`;
+      }
+    });
+    return history;
+  }
+
+  getProgressbarActiveStatus() {
+    if (this.props.loading) {
+      return '';
+    }
+    const client = this.props.client;
+    const lastStatus = this.getLastStatus(client.referralStatusHistory);
+    let cssClass = 'default';
+    if (lastStatus) {
+      cssClass = ReferralStatusList[lastStatus.status].cssClass;
+    } else if (client.matchingScore) {
+      cssClass = ReferralStatusList[0].cssClass;
+    }
+    return 'progress-bar progress-bar-striped active progress-bar-' + cssClass;
+  }
+
+  getCurrentReferralStatus() {
+    if (this.props.loading) {
+      return '';
+    }
+    const client = this.props.client;
+    const lastStatus = this.getLastStatus(client.referralStatusHistory);
+    if (lastStatus) return lastStatus.status + 1;
+    if (client.matchingScore) return 1;
+    return 0;
+  }
+
+  getProgressbarWidth() {
+    if (this.props.loading) {
+      return '';
+    }
+    const client = this.props.client;
+    const total = ReferralStatusList.length;
+    const lastStatus = this.getLastStatus(client.referralStatusHistory);
+    let status = -1;
+    if (lastStatus) {
+      status = lastStatus.status;
+    } else if (client.matchingScore) {
+      status = 0;
+    }
+    const sty = ((status + 1) / total) * 100;
+    return sty + '%';
+  }
+
+  isReferralStatusActiveButton(step) {
+    if (this.props.loading) {
+      return '';
+    }
+    const client = this.props.client;
+    const lastStatus = this.getLastStatus(client.referralStatusHistory);
+    if (lastStatus) {
+      if (step <= lastStatus.status)
+        return 'btn btn-sm btn-arrow-right js-open-referral-status-modal js-tooltip btn-' + ReferralStatusList[step].cssClass;
```

Comment:
can you please extract all static classes (shared in return strings) to a variable. To make it more robust in case it will be changed later